api/v1alpha1/package_types.go

```diff
@@ -70,7 +71,11 @@ type PackageStatus struct {
 
 	// +kubebuilder:validation:Required
 	// Version currently installed
-	Version string `json:"version"`
+	CurrentVersion string `json:"currentVersion"`
+
+	// +kubebuilder:validation:Required
+	// Version to be installed
+	TargetVersion string `json:"targetVersion"`
```

Comment:
You should probably rename PackageVersion and use that. The user shouldn't be messing with status

Reply:
This isn't modified by the user, this is only what the controller reports as being the target version. It will be equal to the spec's `PackageVersion` unless we have no version in which case it will be `latest` (pending package upgrade discussion).